completions/ip

```diff
@@ -192,14 +192,14 @@ _ip()
                     ;;
                 a|add|d|del|change|append|r|replace|monitor)
                     if [[ $prev == via ]]; then
-                        COMPREPLY=( $(compgen -W "$(ip r|grep -oP \
-                        '(?<=via )[0-9.]*'|sort -u)" -- "$cur") )
+                        COMPREPLY=( $(compgen -W "$($1 r|command sed -ne \
+                        's/.*via \([0-9.]*\).*/\1/p')" -- "$cur") )
                     elif [[ $prev == $subcmd ]]; then
                         COMPREPLY=( $(compgen -W "table default \
-                        $(ip r|cut -d ' ' -f 1|sort -u)" -- "$cur") )
+                        $($1 r|command cut -d ' ' -f 1)" -- "$cur") )
                     elif [[ $prev == dev ]]; then
-                        COMPREPLY=( $(compgen -W "$(ip r|grep -oP \
-                        '(?<=dev )[a-z0-9-]*'|sort -u)" -- "$cur") )
+                        COMPREPLY=( $(compgen -W "$($1 r|command sed -ne \
+                        's/.*dev \([a-z0-9-]*\).*/\1/p')" -- "$cur") )
```

Comment:
Shouldn't we complete `dev` with `_available_interfaces` (maybe with `-a`) instead?

Reply:
It could be more correct but too verbose for me (as it includes interfaces that are hardly used for routing like `lo`, and also interfaces that are not even up currently (even with the "-a" switch). I have changed to use `_available_interfaces -a` though.